### DISCUSSION TITLE
404 GET error using visualization

### DISCUSSION THREAD

**Discussion by NikkiKamphuis** [mesa/mesa#1801] [Q&A]
Hi all!

I have been working on my thesis using my predecessors code. While his code for visualization of the model works just fine on my laptop, my own code provides the following error:

404 GET /local/car_empty_safe (127.0.0.1) 1.00ms
404 GET /local/drone_empty_fast (127.0.0.1) 2.00ms
404 GET /local/drone_empty_safe (127.0.0.1) 2.00ms
404 GET /local/White,Grey,Grey (127.0.0.1) 2.99ms
404 GET /local/Red,Grey,Grey (127.0.0.1) 2.99ms

I receive the same type of error, when I create a server that does not use any custom images.

I am using the same file structure and only altered the model and agent classes. Does anyone know where I should look for the problem?

Thanks in advance!

(The repository is also shared on my account)

**Comment by rht** [comment]
This is probably the reason: https://github.com/NikkiKamphuis/Thesis/blob/87c2ed01bdb2a0b076933ba5c4896f5c1e0737ef/!_Model/Agent_based_model/Local_Mesa/Visualisation/templates/js/CustomGridDraw.js#L305. you are passing shape information instead of an actual filename.

Not sure how much time left you have for your thesis, but we have a new Solara-based frontend that is much simpler to develop. Everything is in Python and Matplotlib (https://mesa.readthedocs.io/en/stable/tutorials/visualization_tutorial.html). The entire visualization code is in a single file https://github.com/projectmesa/mesa/blob/main/mesa/experimental/jupyter_viz.py, that is ~300 LOC. I haven't supported custom images grid yet, but it amounts to drawing scatter plot in Matplotlib with image files.

**Reply by NikkiKamphuis** [reply]
I have some time to look into that, thanks a lot!
